Detectors/TPC/calibration/SpacePoints/src/TrackInterpolation.cxx

```diff
@@ -252,19 +255,29 @@ void TrackInterpolation::process()
   trackIndices.insert(trackIndices.end(), mTrackIndices[mTrackTypes[GTrackID::ITSTPC]].begin(), mTrackIndices[mTrackTypes[GTrackID::ITSTPC]].end());
 
   int nSeeds = mSeeds.size();
-  int maxOutputTracks = (mMaxTracksPerTF >= 0) ? mMaxTracksPerTF + mAddTracksITSTPC : nSeeds;
+  int maxOutputTracks = (mMaxTracksPerTF >= 0) ? mMaxTracksPerTF + mAddTracksForMapPerTF : nSeeds;
   mTrackData.reserve(maxOutputTracks);
   mClRes.reserve(maxOutputTracks * param::NPadRows);
   bool maxTracksReached = false;
   for (int iSeed = 0; iSeed < nSeeds; ++iSeed) {
-    if (mMaxTracksPerTF >= 0 && mTrackDataCompact.size() >= mMaxTracksPerTF + mAddTracksITSTPC) {
+    if (mMaxTracksPerTF >= 0 && mTrackDataCompact.size() >= mMaxTracksPerTF) {
       LOG(info) << "Maximum number of tracks per TF reached. Skipping the remaining " << nSeeds - iSeed << " tracks.";
       break;
     }
     int seedIndex = trackIndices[iSeed];
     if (mParams->enableTrackDownsampling && !isTrackSelected(mSeeds[seedIndex])) {
       continue;
     }
+    if (!mSingleSourcesConfigured && !mSourcesConfiguredMap[mGIDs[seedIndex].getSource()]) {
+      auto src = findValidSource(mSourcesConfiguredMap, static_cast<GTrackID::Source>(mGIDs[seedIndex].getSource()));
+      if (src == GTrackID::ITSTPCTRD || src == GTrackID::ITSTPC) {
```

Comment:
@martenole , thanks a lot for the implementation. Could you briefly explain this check?

Reply:
In case there are different track sources configured for the residual extraction and the current track type is not configured as valid source for residual extraction for the map we still want to process the valid fraction of it. E.g. we process a ITS-TPC-TRD track from which we extract residuals for vDrift then we still want to process the ITS-TPC part to obtain residuals for the map extraction.
I realized that the implementation is not yet fully done. The loop over the number of seeds must not break if `mMaxTracksPerTF` is reached, but instead still continue to not drop possible tracks until `mAddTracksForMapPerTF+mMaxTracksPerTF` is reached. I will update the PR, but I had some build problems and might have it only in the afternoon